src/main/java/frc/robot/RobotContainer.java

```diff
@@ -140,6 +140,9 @@ private void configureButtonBindings() {
 
         final JoystickButton shootFull = new JoystickButton(buttonBox, 4);
         shootFull.whileHeld(new Shoot(1.0, m_shooter));
+
+        final JoystickButton shootJoystickSpeed = new JoystickButton(buttonBox, 4);
+        shootJoystickSpeed.whileHeld(new Shoot(buttonBox.getX(), m_shooter));
```

Comment:
Not sure if this works. If it does, awesome. If not, then `buttonBox::getX` might work instead. See [this documentation](https://docs.wpilib.org/en/stable/docs/software/commandbased/convenience-features.html#method-references-java).